internal/elasticsearch/pipeline/processors.go

```diff
@@ -0,0 +1,186 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipeline
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+type Processor struct {
```

Comment:
Do you see feasible adding unit tests for this method? I see a lot of calculations that would benefit for the tests

Reply:
Done